lib/streamlit/elements/widgets/file_uploader.py

```diff
@@ -262,9 +262,16 @@ def file_uploader(
             .. |st.markdown| replace:: ``st.markdown``
             .. _st.markdown: https://docs.streamlit.io/develop/api-reference/text/st.markdown
 
-        type : str or list of str or None
-            Array of allowed extensions. ['png', 'jpg']
-            The default is None, which means all extensions are allowed.
+        type : str, list of str, or None
+            The allowed file types. This can be one of the following:
+
+            - None, which allows all file types to be uploaded.
+            - ``"image"``, ``"audio"``, ``"video"``, or ``"text"`` to allow all files of
+              that type. Note that different browsers may allow different file
+              extensions for these types.
```

Comment:
Would it be correct to refer to MIME type here and point people to https://mimetype.io/all-types for a comprehensive list? Or do we have something else happening in-between at the browser level to make that incorrect?